main/broadcasters.c

```diff
@@ -548,6 +561,25 @@ static void mijia_sensor_metadata_get(uint8_t *adv_data, size_t adv_data_len,
                 le16toh(*(uint16_t *)(mijia_data_entry->data + 2)) / 10.0);
             cb("Humidity", s, ctx);
         }
+        else if (mijia_data_entry->data_type == MIJIA_SENSOR_DATA_TYPE_MOTION_LIGHT)
+        {
+            uint32_t val = mijia_data_entry->data[0]
+                | (mijia_data_entry->data[1] << 8) | (mijia_data_entry->data[2] << 16);
+            if (device_type == MIJIA_DEVICE_TYPE_CGPR1)
+            {
+                sprintf(s, "%u", val);
+                cb("Illuminance", s, ctx);
+            }
+            else if (device_type == MIJIA_DEVICE_TYPE_MJYD02YL)
+            {
+                cb("Light", val == 100 ? "1" : "0", ctx);
+            }
+            else if (device_type == MIJIA_DEVICE_TYPE_RTCGQ02LM)
+            {
+                cb("Light", val == 256 ? "1" : "0", ctx);
+            }
+            cb("Motion", "1", ctx);
```

Comment:
So, if the data type is `MIJIA_SENSOR_DATA_TYPE_MOTION_LIGHT` then motion is always detect? How is motion reset? Will the new message be with data type `MIJIA_SENSOR_DATA_TYPE_MOTION` from the same device?

Reply:
No, there is just no reset timer on the device. You can only reset to zero inside your home automation software, e.g. with the "expire" attribute in openHAB.

Reply:
At least on the RTCGQ02LM the light is also broadcast separately as a `MIJIA_SENSOR_DATA_TYPE_LIGHT` message

> publishing motion twice, in succession, 0 and then 1

Not necessary IMO

> perhaps we should publish `true`/`false`

ehh, there's plenty of other types that already report a byte directly that are 0/1, let's not introduce inconsistencies

Reply:
OK, I'm with you